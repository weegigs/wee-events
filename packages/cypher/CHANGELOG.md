# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.18.0 (2023-02-24)

### Bug Fixes

- correct depenencies ([97dfaab](https://github.com/weegigs/wee-events/commit/97dfaab70f4863b71b190f6be68bc75e72618803))
- error when destructuring reducers for unknown events ([f0d08df](https://github.com/weegigs/wee-events/commit/f0d08dffa683f331e6f9c1d191e51a7ecce86b4f))
- removed #private fields ([dd828f9](https://github.com/weegigs/wee-events/commit/dd828f92771d11b63e2bf30184146d8dace00fff))
- switch references from @weegigs/wee-events to @weegigs/events-core ([f3b5225](https://github.com/weegigs/wee-events/commit/f3b522550cc0f7f11a967893d6be61c716e08d4e))

### Features

- effects based service definition ([59fbe04](https://github.com/weegigs/wee-events/commit/59fbe0433839f220f5f9cc5aa43d0dc78e7c0c19))
