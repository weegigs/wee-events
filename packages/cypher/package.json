{
  "name": "@weegigs/events-cypher",
  "version": "0.16.0",
  "description": "Cypher Utilities for Wee Events",
  "author": "Kevin O'Neill <kevin@weegigs.com.au>",
  "license": "MIT",
  "main": "./lib/index.js",
  "types": "./lib/index.d.ts",
  "packageManager": "yarn@3.2.1",
  "repository": {
    "type": "git",
    "url": "https://github.com/weegigs/wee-events.git"
  },
  "scripts": {
    "build": "yarn ws:build",
    "clean": "yarn ws:clean",
    "compile": "yarn ws:compile",
    "test": "yarn ws:test",
    "lint": "yarn ws:lint",
    "doctor": "yarn ws:doctor",
    "fix": "yarn run ws:fix",
    "prepack": "yarn run ws:prepack",
    "publish": "yarn run ws:publish"
  },
  "peerDependencies": {
    "@aws-sdk/lib-dynamodb": "^3.54.0",
    "@weegigs/events-core": "workspace:^"
  },
  "dependencies": {
    "@aws-sdk/lib-dynamodb": "^3.172.0",
    "@aws-sdk/smithy-client": "^3.171.0"
  },
  "devDependencies": {
    "@aws-sdk/client-dynamodb": "^3.172.0",
    "@aws-sdk/lib-dynamodb": "^3.172.0",
    "@aws-sdk/smithy-client": "^3.171.0",
    "@aws-sdk/types": "^3.171.0",
    "@types/jest": "^29.0.3",
    "@types/node": "^16.11.59",
    "@weegigs/events-core": "workspace:*",
    "jest": "^29.0.3",
    "testcontainers": "^8.14.0",
    "ts-jest": "^29.0.1",
    "typescript": "^4.8.3"
  },
  "files": [
    "lib"
  ],
  "jest": {
    "preset": "ts-jest",
    "testEnvironment": "node",
    "collectCoverageFrom": [
      "**/*.{ts,tsx}",
      "!**/node_modules/**"
    ],
    "testMatch": [
      "**/*.spec.ts",
      "**/*.test.ts"
    ],
    "collectCoverage": false,
    "coverageReporters": [
      "html"
    ]
  },
  "prettier": {
    "printWidth": 120,
    "trailingComma": "es5"
  }
}
