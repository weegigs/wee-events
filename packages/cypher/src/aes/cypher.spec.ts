import { describe, it, expect } from "vitest";
import { AESCypher } from "./cypher";

describe("aes cypher", () => {
  const id = { key: "test-id", type: "test-type" };
  const payload = { test: "test" };

  it("should encrypt payloads", async () => {
    const cypher = AESCypher.create({ nonce: () => Buffer.from("123456789012") });
    const data = await cypher.encrypt(id, payload);
    expect(data).toBeDefined();
    expect(data).toMatchSnapshot();
  });

  it("should round trip payloads", async () => {
    const cypher = AESCypher.create({ nonce: () => Buffer.from("123456789012") });

    const data = await cypher.encrypt(id, payload);
    const recovered = await cypher.decrypt(id, data);

    expect(recovered).toEqual(payload);
  });
});
