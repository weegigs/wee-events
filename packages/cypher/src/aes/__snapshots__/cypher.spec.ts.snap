// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aes cypher should encrypt payloads 1`] = `
{
  "data": [
    123,
    34,
    110,
    111,
    110,
    99,
    101,
    34,
    58,
    34,
    77,
    84,
    73,
    122,
    78,
    68,
    85,
    50,
    78,
    122,
    103,
    53,
    77,
    68,
    69,
    121,
    34,
    44,
    34,
    100,
    97,
    116,
    97,
    34,
    58,
    34,
    114,
    49,
    48,
    72,
    50,
    121,
    85,
    98,
    73,
    118,
    111,
    114,
    110,
    48,
    106,
    76,
    76,
    106,
    105,
    99,
    34,
    44,
    34,
    116,
    97,
    103,
    34,
    58,
    34,
    118,
    97,
    72,
    69,
    121,
    71,
    115,
    120,
    73,
    107,
    99,
    78,
    72,
    105,
    50,
    69,
    111,
    77,
    100,
    85,
    56,
    81,
    61,
    61,
    34,
    125,
  ],
  "type": "Buffer",
}
`;
